nucypher/blockchain/eth/actors.py

```diff
@@ -666,6 +515,127 @@ def set_min_reward_rate_range(self,
         return receipt
 
 
+class Allocator:
+    def __init__(self, filepath: str, registry, deployer_address):
+
+        self.log = Logger("allocator")
+        self.staking_agent = ContractAgency.get_agent(StakingEscrowAgent,
+                                                      registry=registry)  # type: StakingEscrowAgent
+        self.max_substakes = self.staking_agent.contract.functions.MAX_SUB_STAKES().call()
+
+        with open(filepath, 'r') as allocation_file:
```

Comment:
Perhaps move the disk I/O to a method (which will make testing/mocking a bit easier)

Reply:
✅